crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -198,6 +200,10 @@ impl CumulocityConverter {
         let service_type = config.service_type.clone();
         let c8y_host = config.c8y_host.clone();
         let cfg_dir = config.config_dir.clone();
+        let tedge_config_location = TEdgeConfigLocation::from_custom_root(&cfg_dir);
+        let config_repository = TEdgeConfigRepository::new(tedge_config_location);
+        let tedge_config = config_repository.load()?;
+        let topic_root = tedge_config.mqtt.topic_root.to_string();
```

Comment:
No this is useless and error prone as the c8y related part of the config has been already read and passed via the `config: C8yMapperConfig` argument.

The `mqtt.topic_root` is indeed missing from the later. It must be added there.

By the way this new config parameter will have to be used to fix the converter `mqtt_schema` which is [incorrectly initialized with a default value](https://github.com/thin-edge/thin-edge.io/blob/main/crates/extensions/c8y_mapper_ext/src/converter.rs#L246)

Reply:
fixed